services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchRulesSpec.groovy

```diff
@@ -115,49 +116,79 @@ class SwitchRulesSpec extends BaseSpecification {
     }
 
     @Unroll
-    def "Able to delete #data.description rules from a switch"() {
+    def "Able to delete rules from a switch (delete-action=#data.deleteRulesAction)"() {
         given: "A switch with some flow rules installed"
         def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
         flowHelper.addFlow(flow)
 
-        when: "Delete #data.description rules from the switch"
+        when: "Delete rules from the switch"
+        def flowRules = getFlowRules(srcSwitch.dpId)
         def deletedRules = northboundService.deleteSwitchRules(srcSwitch.dpId, data.deleteRulesAction)
+        def expectedRules = srcSwDefaultRules
+        switch (data.deleteRulesAction) {
```

Comment:
I don't really like all these `if`s. Test is getting too complicated. You can put the `expectedRules` variable inside your `where` block instead of calculating it for different cases inside the test. This will make your `where` block as well as test itself to be more readable, because you will have a clear expected result directly in where for every delete action(every iteration), instead of calculating it in two different places in test for different scenarios. Since you cannot directly calculate it in 'where'(because you don't know the exact flowRules), you can supply a 'getExpectedRules' closure in the `where`, that will be called during test execution and take current flowRules as param (similar to `"Unable to create flow with conflicting vlans or flowIds"` test)   
Also, why on line 142, for the DROP_ALL action, you can't compare actualRules with an empty array? For DROP_ALL the expectedRules is an empty array, isn't it? I mean it would be at least less painful if we have all our `if`s in one place (inside this `switch`) instead of having multiple places where we split the logic. In this case you can completely remove the `rulesRemained` variable as well, because you will run 'compareRules' for all cases and it takes size into account
I don't heavily insist on these changes, just shared my point of view

Reply:
Fixed.